config/config.go

```diff
@@ -26,6 +26,9 @@ type Config struct {
 		AllowUppercase bool   `yaml:"allow_uppercase"`
 		Verbose        bool   `yaml:"verbose"`
 	} `yaml:"r10k"`
+	Pipeline struct {
```

Comment:
should probably add an option for `Enabled` as well. This way admins can ignore webhook pushes from workflows if they want to. Especially in a situation where the pipeline/workflow is maintained by a different team from those admining the webhook server. 

This also allows us to skip the complex logic in the controllers unless pipelines/workflows are enabled.

Reply:
Hmm wouldn't that make sense for push event as well? More like a `Strategy` option were you can select which event will cause a deployment of Puppet code.

Because I don't think you want to listen for both events as `push` will always deploy before `pipeline`.

Reply:
That depends on how/when the user wants the pipeline to trigger and how their repository is setup. Generally, once a pull request is merged on GitHub, you don't want/need the workflow to run again since the first run is run against the PR merged into a detached master branch. 

Merging a Pull Request/Merge Request generates a push event itself, and generally you'll want the PR/MR to either auto-merge on passing tests or not allow merging until tests pass. Which will then generate a push event (and a PR/MR closed event, IIRC). Rarely are post-merge pipelines acted on (at least in my experience). So deploying on push events are always the default anyway.